### ISSUE TITLE
Please rebuild graphviz for libffi 3.3  (required for Python 3.9)

### ISSUE THREAD

**Issue by rolweber** [ContinuumIO/anaconda-issues#12030]
Hello Team Anaconda,

recent builds of `python` require `libffi` 3.3. The latest available build of `graphviz` for Linux still requires `libffi` 3.2.1.

Could you please rebuild `graphviz` for `libffi` 3.3?

Thanks and cheers!

```txt
# conda search graphviz=2.40.1=h21bd128_2 --info
Loading channels: done
graphviz 2.40.1 h21bd128_2
--------------------------
file name   : graphviz-2.40.1-h21bd128_2.tar.bz2
name        : graphviz
version     : 2.40.1
build string: h21bd128_2
build number: 2
size        : 6.9 MB
constrains  : ()
license     : EPL v1.0
subdir      : linux-64
url         : https://repo.anaconda.com/pkgs/main/linux-64/graphviz-2.40.1-h21bd128_2.tar.bz2
md5         : 05d3caefd92848b6b5db824671d8873b
dependencies: 
  - cairo >=1.14.12,<2.0a0
  - expat >=2.2.5,<3.0a0
  - fontconfig >=2.13.0,<3.0a0
  - freetype >=2.9.1,<3.0a0
  - fribidi >=1.0.4,<2.0a0
  - glib >=2.56.1,<3.0a0
  - graphite2 >=1.3.11,<2.0a0
  - harfbuzz >=1.7.6,<2.0a0
  - icu >=58.2,<59.0a0
  - jpeg >=9b,<10a
  - libffi >=3.2.1,<3.3a0
  - libgcc-ng >=7.2.0
  - libpng >=1.6.34,<1.7.0a0
  - libtiff >=4.0.9,<5.0a0
  - libuuid >=1.0.3,<2.0a0
  - libxml2 >=2.9.8,<2.10.0a0
  - pango >=1.42.1,<2.0a0
  - pcre >=8.42,<9.0a0
  - pixman >=0.34.0,<1.0a0
  - xz >=5.2.4,<6.0a0
  - zlib >=1.2.11,<1.3.0a0
```

**Comment by rolweber** [comment]
ping @anaconda-pkg-build

**Comment by awilliams-anaconda** [comment]
@rolweber which version of Python are you on?

**Comment by rolweber** [comment]
I'm building container images with Python 3.8, 3.7 and (still) 3.6. The latest Anaconda builds of all three versions require `libffi` 3.3.

```txt
Loading channels: done
python 3.8.5 h7579374_1
-----------------------
file name   : python-3.8.5-h7579374_1.conda
name        : python
version     : 3.8.5
build       : h7579374_1
build number: 1
size        : 49.3 MB
license     : PSF
subdir      : linux-64
url         : https://repo.anaconda.com/pkgs/main/linux-64/python-3.8.5-h7579374_1.conda
md5         : fa19bb6876427f0aadd3e2c2804fbb61
timestamp   : 2020-09-04 07:36:03 UTC
dependencies: 
  - ld_impl_linux-64
  - libffi >=3.3,<3.4.0a0
  - libgcc-ng >=7.3.0
  - ncurses >=6.2,<7.0a0
  - openssl >=1.1.1g,<1.1.2a
  - readline >=8.0,<9.0a0
  - sqlite >=3.33.0,<4.0a0
  - tk >=8.6.10,<8.7.0a0
  - xz >=5.2.5,<6.0a0
  - zlib >=1.2.11,<1.3.0a0
  - pip
```
```txt
python 3.7.9 h7579374_0
-----------------------
file name   : python-3.7.9-h7579374_0.conda
name        : python
version     : 3.7.9
build       : h7579374_0
build number: 0
size        : 45.3 MB
license     : PSF
subdir      : linux-64
url         : https://repo.anaconda.com/pkgs/main/linux-64/python-3.7.9-h7579374_0.conda
md5         : e8871e5a420f47397833fd2f3f0dac7a
timestamp   : 2020-08-31 12:50:58 UTC
dependencies: 
  - ld_impl_linux-64
  - libffi >=3.3,<3.4.0a0
  - libgcc-ng >=7.3.0
  - ncurses >=6.2,<7.0a0
  - openssl >=1.1.1g,<1.1.2a
  - readline >=8.0,<9.0a0
  - sqlite >=3.33.0,<4.0a0
  - tk >=8.6.10,<8.7.0a0
  - xz >=5.2.5,<6.0a0
  - zlib >=1.2.11,<1.3.0a0
  - pip
```
```txt
Loading channels: done
python 3.6.12 hcff3b4d_2
------------------------
file name   : python-3.6.12-hcff3b4d_2.conda
name        : python
version     : 3.6.12
build       : hcff3b4d_2
build number: 2
size        : 29.7 MB
license     : PSF
subdir      : linux-64
url         : https://repo.anaconda.com/pkgs/main/linux-64/python-3.6.12-hcff3b4d_2.conda
md5         : 5a6b4f2bf00e5f83ff0aa7365e4618fb
timestamp   : 2020-09-08 23:14:17 UTC
dependencies: 
  - ld_impl_linux-64
  - libffi >=3.3,<3.4.0a0
  - libgcc-ng >=7.3.0
  - libstdcxx-ng >=7.3.0
  - ncurses >=6.2,<7.0a0
  - openssl >=1.1.1g,<1.1.2a
  - readline >=8.0,<9.0a0
  - sqlite >=3.33.0,<4.0a0
  - tk >=8.6.10,<8.7.0a0
  - xz >=5.2.5,<6.0a0
  - zlib >=1.2.11,<1.3.0a0
  - pip
```

**Comment by rolweber** [comment]
As I am under pressure to install the security fixes in the latest python builds, I had to get creative. Luckily, my users install their Python API from PyPI's `graphviz` rather than conda's `python-graphviz`. Therefore, I need to provide the `dot` command from Anaconda's `graphviz` in the user environment, but not the package itself.

As a workaround, I'm now installing the Anaconda `graphviz` into a dedicated environment, and I symlink the `dot` command into the PATH. The dedicated environment doesn't have to be consistent, so I can force-remove the vulnerable python package from there. While trying different things, I also installed the `graphviz` package with `--no-deps` into an environment that had `libffi` 3.3, and it worked anyway. Unfortunately, the `--no-deps` trick worked only with conda 4.6. For conda 4.8, I had to download the package and patch `info/index.json` to drop the version spec from the `libffi` dependency. Overall, it's simpler to install with the declared dependencies and then remove `python`. Still, all of these are ugly hacks. Unfortunately, the Linux distributions I have to use do not provide `graphviz` in their repositories.

I trust that Anaconda will not force the users of their `graphviz` package to stay on vulnerable `python` versions for much longer. So I'm looking forward to get rid of the workaround.

**Comment by rolweber** [comment]
For the records, the oldest available build of `graphviz` from Anaconda doesn't restrict `libffi` and can be installed with current builds of `python`. But it requires the backlevel `freetype` 2.8, so it doesn't fit into environments with the more recent 2.9 and 2.10 versions of `freetype`.

**Comment by efussi** [comment]
We need this update for the **osx-64** platform as well where we see the same issue with **graphviz** v2.40.1.

We don't see the issue with **win-64** because there only an older version (v2.38) of **graphviz** is available.

**Comment by rolweber** [comment]
Resolved with graphviz 2.50.0.

Thank you!
